modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/jdbc/JdbcIndexMeta.java

```diff
@@ -58,26 +60,20 @@ public class JdbcIndexMeta implements JdbcRawBinarylizable {
     }
 
     /**
-     * @param schemaName Schema name.
-     * @param tblName Table name.
-     * @param idx Index info.
+     * @param idx Index descriptor.
      */
-    JdbcIndexMeta(String schemaName, String tblName, GridQueryIndexDescriptor idx) {
-        assert tblName != null;
-        assert idx != null;
-        assert idx.fields() != null;
-
-        this.schemaName = schemaName;
-        this.tblName = tblName;
-
+    JdbcIndexMeta(IndexDescriptor idx) {
+        schemaName = idx.table().type().schemaName();
+        tblName = idx.table().type().tableName();
         idxName = idx.name();
+
         type = idx.type();
-        fields = new ArrayList(idx.fields());
 
-        fieldsAsc = new ArrayList<>(fields.size());
+        fields = new ArrayList<>(idx.keyDefinitions().keySet());
 
-        for (int i = 0; i < fields.size(); ++i)
-            fieldsAsc.add(!idx.descending(fields.get(i)));
+        fieldsAsc = idx.keyDefinitions().values().stream()
+            .map(d -> d.order().sortOrder() == SortOrder.ASC)
```

Comment:
Filling `fields` and `fieldAsc` must be done within single stream. I'm not sure that order for `keySet()` and `values()` are guaranteed.